### ISSUE TITLE
Reproducing Results

### ISSUE THREAD

**Issue by Quest2GM** [RobotiXX/multimodal-fusion-network#3]
Hi @bhaba-ranjan,

Thanks for sharing the repository, I'm currently looking to reproduce this work. Are the default hyperparameters that are set in the multimodal.py file enough to reproduce your model? Also, do you have checkpoints for your best model that you can make public?

Thanks,
Quest2GM

**Comment by Quest2GM** [comment]
As a followup, I trained my first model with the default parameters. These are the error graphs I was able to produce:

![image](https://github.com/RobotiXX/multimodal-fusion-network/assets/39883561/19cc2359-5053-45ec-8ba6-658a1bb271ea)
![image](https://github.com/RobotiXX/multimodal-fusion-network/assets/39883561/db684b57-9509-49b6-bb25-6b3716bd6589)

The path prediction graphs seem to be doing well, and the predictions are also pretty decent upon review. However, I noticed that for the velocity prediction, the curve completely flattens out after roughly the 50th epoch, and the loss curve doesn't go below that. After looking at the output of the networks for velocity, the velocity prediction is the same every time regardless of the input (1.6 m/s and -0.55rad/s). Is there any reason for this? Did the authors come across this as well during training?

Thanks for any insight.

Siddarth

**Comment by bhaba-ranjan** [comment]
It should be enough to reproduce the result but make sure you are on the default branch which is `end-to-end-2-multimodal-v2`. You need to do some tuning for the physical experiment though.  

The same output makes me think if there is a imbalance in your training data. Try to sample it to make it balanced. Also, you can change the network a bit to learn angular velocity only.

**Comment by Quest2GM** [comment]
Thanks for your response @bhaba-ranjan 

Do you have the checkpoints of the model that you had trained, and it is possible to make it available? It would be nice to have something to compare my current model against.

Also what tuning did you have to do for the physical experiment?

**Comment by bhaba-ranjan** [comment]
I have shared it on your UT mail.

**Comment by Quest2GM** [comment]
Hi @bhaba-ranjan,

Thank you for sharing this. After importing the weights you sent, I had one quick clarification to the code for these weights. The raw output of the model for the angular velocity is on the order of 10^4. I'm wondering what was the scale that you used to convert back to angular velocity. Specifically, I'm wondering what are the interpolation bounds that you used: https://github.com/RobotiXX/multimodal-fusion-network/blob/end-to-end-2-multimodal-v2/scripts/data_builder/cmd_scaler.py#L16C5-L16C84, <=  the outputs of the model are not on the orders of 10^7 as given in this line.

Thanks,
Siddarth
